wallet-api/src/Wallet/API.hs

```diff
@@ -273,13 +275,18 @@ otherError = throwError . OtherError
 createPayment :: (Functor m, WalletAPI m) => Value -> m (Set.Set TxIn')
 createPayment vl = fst <$> createPaymentWithChange vl
 
--- | Transfer some funds to an address locked by a script.
-payToScript :: (Monad m, WalletAPI m) => Address' -> Value -> DataScript -> m Tx
-payToScript addr v ds = do
+-- | Transfer some funds to an address locked by a script, returning the 
+--   transaction that was submitted.
+payToScript' :: (Monad m, WalletAPI m) => Address' -> Value -> DataScript -> m Tx
+payToScript' addr v ds = do
```

Comment:
isn't `payToScript_` more idiomatic in Haskell?

Reply:
You mean for the one that returns unit, right? I agree that's more idiomatic, although it's a bit odd that perhaps you will normally be using the `_` variant.

Reply:
ok I'll change it, I did that to make the standard version (returning unit) look nice in the examples but it's probably better to stick with the convention